src/pages/BookAppointment.tsx

```diff
@@ -23,20 +25,39 @@ const TIME_SLOTS = [
 ];
 
 export default function BookAppointment() {
+  const navigate = useNavigate();
   const [isLoading, setIsLoading] = useState(false);
-  const [selectedDate, setSelectedDate] = useState('');
-  const [selectedTime, setSelectedTime] = useState('');
-  const [selectedDepartment, setSelectedDepartment] = useState('');
+  const [formData, setFormData] = useState({
+    fullName: '',
+    email: '',
+    phone: '',
+    department: '',
+    appointmentDate: '',
+    appointmentTime: '',
+    symptoms: '',
+  });
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>) => {
+    const { name, value } = e.target;
+    setFormData(prev => ({
+      ...prev,
+      [name]: value
+    }));
+  };
 
   const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     setIsLoading(true);
 
-    // Simulate API call
-    await new Promise((resolve) => setTimeout(resolve, 1000));
-    
-    toast.success('Appointment booked successfully!');
-    setIsLoading(false);
+    try {
```

Comment:
Updating the Mongo DB with proper appointment data, and adding the dashboard routes